app/models/relationship.rb

```diff
@@ -17,10 +17,16 @@ def filtered?(of_type, except_type)
   end
 
   def self.filter_by_resource_type(relationships, options)
-    of_type = options[:of_type].to_miq_a
-    except_type = options[:except_type].to_miq_a
+    of_type = Array.wrap(options[:of_type])
+    except_type = Array.wrap(options[:except_type])
     return relationships if of_type.empty? && except_type.empty?
-    relationships.reject { |r| r.filtered?(of_type, except_type) }
+    if relationships.kind_of?(Array)
+      relationships.reject { |r| r.filtered?(of_type, except_type) }
+    else
+      relationships = relationships.where(:resource_type => of_type) if of_type.present?
+      relationships = relationships.where.not(:resource_type => except_type) if except_type.present?
```

Comment:
The purpose of this method `filter_by_resource_type` is to filter on these 2 fields.

You're proposing a method that just does the in memory part?

of note, `relationships.reject { |r| r.filtered?(of_type, except_type) }` is only used in this 1 spot.
Also, of note, `:except_type` is only ever used in 1 spot of our code.